src/Umbraco.Web.Website/ActionResults/RedirectToUmbracoUrlResult.cs

```diff
@@ -31,6 +42,11 @@ public Task ExecuteResultAsync(ActionContext context)
 
         var destinationUrl = _umbracoContext.OriginalRequestUrl.PathAndQuery;
 
+        if (_queryString.HasValue)
+        {
+            destinationUrl = _umbracoContext.OriginalRequestUrl.AbsolutePath + _queryString.ToUriComponent();
+        }
+
```

Comment:
This maybe needs a comment, as it seems a little unintuitive to me.

Given we are using `OriginalRequestUrl.PathAndQuery` that presumably should include the querystring already.  So why do we need to pass it in?

If we do, then I guess it's because we are allowing someone to override the original query string with a different one - but in that case, maybe it's better to just expect them to do `Redirect` directly, and have full control over the URL that's used?

So just trying to understand the use case for this, and if you do believe there's a real one, maybe you could add in a comment for future reference?  Thanks.